File: benchmarks/oopsla25/subjects/450.py
#!/usr/bin/env python3

import sys

def f450(s: str):
  i = 0
  while i < len(s):
    assert s[i+0] == "a"
    assert s[i+1] != "a"
    assert s[i+1] != "b"
    assert s[i+2] == "b"
    i += 3

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f450(my_string)

File: benchmarks/oopsla25/subjects/450.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f450(const char *s) {
    int i = 0;
    while (i < strlen(s)) {
        assert(s[i + 0] == 'a');
        assert(s[i + 1] != 'a' && s[i + 1] != 'b');
        assert(s[i + 2] == 'b');
        i = i+3;
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f450(my_string);
  return 0;
}
